src/app/routes/api/v3/forms/__tests__/public-forms.verification.routes.spec.ts

```diff
@@ -231,4 +239,322 @@ describe('public-forms.verification.routes', () => {
+    beforeEach(() => {
+      MockTwilio.messages.create.mockResolvedValue({
+        sid: 'mockSid',
+      })
+    })
+
+    it('should return 201 when parameters for email field are valid', async () => {
+      // Act
+      const response = await request
+        .post(
+          `/forms/${mockVerifiableFormId}/fieldverifications/${mockTransactionId}/fields/${mockEmailFieldId}/otp/generate`,
+        )
+        .send({
+          fieldType: VerifiableFieldType.Email,
+          answer: 'open@gov.sg',
+        })
+
+      // Assert
+      expect(response.status).toBe(StatusCodes.CREATED)
+      expect(response.text).toBe(getReasonPhrase(StatusCodes.CREATED))
+    })
+
+    it('should return 201 when parameters for mobile field are valid', async () => {
+      // Act
+      const response = await request
+        .post(
+          `/forms/${mockVerifiableFormId}/fieldverifications/${mockTransactionId}/fields/${mockMobileFieldId}/otp/generate`,
+        )
+        .send({
+          fieldType: VerifiableFieldType.Mobile,
+          // NOTE: a valid mobile number is +65 followed by ANY 8 digits
+          answer: '+6512345678',
+        })
+
+      // Assert
+      expect(response.status).toBe(StatusCodes.CREATED)
+      expect(response.text).toBe(getReasonPhrase(StatusCodes.CREATED))
+    })
+
+    it('should return 400 when fieldType is email but the provided email is not valid', async () => {
+      // Act
+      const response = await request
+        .post(
+          `/forms/${mockVerifiableFormId}/fieldverifications/${mockTransactionId}/fields/${mockEmailFieldId}/otp/generate`,
+        )
+        .send({
+          fieldType: VerifiableFieldType.Email,
+          answer: 'notanemail',
+        })
+
+      // Assert
+      expect(response.status).toBe(StatusCodes.BAD_REQUEST)
+      expect(response.body).toEqual(
+        buildCelebrateError({
+          body: {
+            key: 'answer',
+            message: 'Please ensure that the email provided is valid.',
+          },
+        }),
+      )
+    })
+
+    it('should return 400 when fieldType is mobile but the provided phone number is not valid', async () => {
+      // Act
+      const response = await request
+        .post(
+          `/forms/${mockVerifiableFormId}/fieldverifications/${mockTransactionId}/fields/${mockMobileFieldId}/otp/generate`,
+        )
+        .send({
+          fieldType: VerifiableFieldType.Mobile,
+          // 7 digits after +65 instead of 8
+          answer: '+651234567',
+        })
+
+      // Assert
+      expect(response.status).toBe(StatusCodes.BAD_REQUEST)
+      expect(response.body).toEqual(
+        buildCelebrateError({
+          body: {
+            key: 'answer',
+            message: 'Please ensure that the phone number provided is valid.',
+          },
+        }),
+      )
+    })
+
+    it('should return 400 when the parameters are malformed', async () => {
```

Comment:
Would it be better to compress this test and the expired transaction test into one test since they are testing the same thing? The description could be 'should return 400 when `getOtpData` returns null' because the malformed parameter and expired transaction is detected in `getOtpData` right?

Reply:
this is an error on my end, should've created an expired transaction in the db. good catch!